src/python_testing/TC_SU_2_5.py

```diff
@@ -0,0 +1,509 @@
+        asserts.assert_greater_equal(
+            total_wait_time, delayed_apply_action_time, f"Software Udpate occurred before the defined time of: {delayed_apply_action_time}")
+        software_version_attr_handler.reset()
+        await software_version_attr_handler.cancel()
+        # Verify the version is the same
+        await self.verify_version_applied_basic_information(controller=self.controller, node_id=self.requestor_node_id, target_version=self.expected_software_version)
+        self.terminate_provider()
+        self.restart_requestor(self.controller)
+
+        self.step(5)
+        extra_arguments = ['--applyUpdateAction', 'discontinue']
+        self.start_provider(
+            provider_app_path=self.provider_app_path,
+            ota_image_path=self.ota_image,
+            setup_pincode=self.provider_setup_pincode,
+            discriminator=self.provider_discriminator,
+            port=self.ota_provider_port,
+            kvs_path=self.provider_kvs_path,
+            log_file=self.provider_log,
+            extra_args=extra_arguments,
+        )
+        current_sw_version = await self.read_single_attribute_check_success(
+            dev_ctrl=self.controller,
+            cluster=Clusters.BasicInformation,
+            attribute=Clusters.BasicInformation.Attributes.SoftwareVersion,
+            node_id=self.requestor_node_id)
+        update_state_attr_handler = AttributeSubscriptionHandler(
+            expected_cluster=Clusters.OtaSoftwareUpdateRequestor,
+            expected_attribute=Clusters.OtaSoftwareUpdateRequestor.Attributes.UpdateState
+        )
+        await update_state_attr_handler.start(dev_ctrl=self.controller, node_id=self.requestor_node_id, endpoint=0,
+                                              fabric_filtered=False, min_interval_sec=0, max_interval_sec=5)
+        await self.announce_ota_provider(self.controller, self.provider_node_id, self.requestor_node_id)
+
+        # Wait Until Downlading
+        update_state_match = AttributeMatcher.from_callable(
+            "Waiting Update state is Downloading",
+            lambda report: report.value == Clusters.OtaSoftwareUpdateRequestor.Enums.UpdateStateEnum.kDownloading)
+
+        update_state_attr_handler.await_all_expected_report_matches([update_state_match], timeout_sec=90)
+        # Wait some time to check if is downloading
+        await asyncio.sleep(5)
+        # Verify the default download path and the file size
+        # Read file for /tmp/test.bin should exists and greater than 0
+        ota_file_data = self.get_downloaded_ota_image_info()
+        logger.info(f"Downloaded ota image data {str(ota_file_data)}")
+        asserts.assert_equal(True, ota_file_data['exists'], f"File is was not downloaded  at {ota_file_data['path']}")
+        asserts.assert_greater(ota_file_data['size'], 0, "Downloaded file is still at 0")
+
+        # Applying
+        update_state_match = AttributeMatcher.from_callable(
```

Comment:
Isn't the DUT told to discontinue the update? Why would it still hit the applying state?